### PR TITLE
questions are created, removed and checked new question for not empty input

### PR DESCRIPTION
**Pull request by VeiveHata**
this branch was merged with fixed bootstrap; not all buttons should be able for clickin. will be fixed after building service. deleating adjacent answers in future.

### CODE REVIEW

File: src/app/components/question/question.component.ts [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
-  styleUrls: ['./question.component.css']
+    selector: 'app-question',
+    templateUrl: './question.component.html',
+    styleUrls: ['./question.component.scss']
 })
 export class QuestionComponent {
 
-    @Input() question: Question;
-    @Input() name: string;
+    @Input() question;
```
Comment (FrontendMentor) [inline src/app/components/question/question.component.ts]: add type for this variable

File: src/app/components/question/question.component.ts [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
+    @Input() name;
```
Comment (FrontendMentor) [inline src/app/components/question/question.component.ts]: add type for this variable

File: src/app/components/question/question.component.ts [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+
+    path: string = 'questions';
+    newAnswer: boolean = false;
+
+    constructor(private questionService: QuestionService) { }
+
+    toAnswer() {
```
Comment (FrontendMentor) [inline src/app/components/question/question.component.ts]: add return type for this method

File: src/app/components/answer/answer.component.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 @Component({
-  selector: 'app-answer',
-  templateUrl: './answer.component.html',
-  styleUrls: ['./answer.component.css']
+    selector: 'app-answer',
+    templateUrl: './answer.component.html',
+    styleUrls: ['./answer.component.css']
 })
-export class AnswerComponent implements OnInit {
+export class AnswerComponent {
```
Comment (FrontendMentor) [inline src/app/components/answer/answer.component.ts]: I'm not sure that we need this component.  Can we merge it with AnswersComponent?

File: src/app/components/question/question.component.ts [hunk 5]
```diff
@@ -1,13 +1,37 @@
...
+        this.newAnswer = !this.newAnswer;
+    }
+
+    addAnswer(answer: string, uid: string, questionKey: string): void {
+
```
Comment (FrontendMentor) [inline src/app/components/question/question.component.ts]: remove empty line here

File: src/app/components/questions/questions.component.ts [hunk 6]
```diff
@@ -10,17 +11,41 @@ import { QuestionComponent } from '../question/question.component';
...
             this.questions = questions;
         });
     }
 
-    getName(id: string): string {
-        return this.questionService.getName(id);
+    setName(id: string): string {
+        let name: string;
+
+        for (let user = 0, usersCount = this.users.length; user < usersCount; user ++) {
```
Comment (FrontendMentor) [inline src/app/components/questions/questions.component.ts]: Could you use `for-each loop` here?

File: src/app/components/userLogin/user-login.component.ts [hunk 7]
```diff
@@ -17,21 +17,21 @@ export class UserLoginComponent {
...
+        const inputValue = this.form.value;
 
-    	this.auth.login(inputValue.email, inputValue.password)
-    		.subscribe(
-    			(success: any) => console.log("success"), //this.router.navigate(['/questions']);
-    			(error: Error) => alert(error)
-    		)
+        this.auth.login(inputValue.email, inputValue.password)
+            .subscribe(
+                success => console.log("success"), //this.router.navigate(['/questions']);
```
Comment (FrontendMentor) [inline src/app/components/userLogin/user-login.component.ts]: add type for `success`

File: src/app/components/userLogin/user-login.component.ts [hunk 8]
```diff
@@ -17,21 +17,21 @@ export class UserLoginComponent {
...
+                error => alert(error)
```
Comment (FrontendMentor) [inline src/app/components/userLogin/user-login.component.ts]: add type for `error`

File: src/app/services/question.service.ts [hunk 9]
```diff
@@ -13,35 +14,72 @@ export class QuestionService {
...
 
-  constructor(private db: AngularFireDatabase) {
-    this.questions = db.list('questions').snapshotChanges().map((changes: any[])  => {
-        return changes.map((questionUnit: any) => ({ key: questionUnit.payload.key, ...questionUnit.payload.val() }));
-    });
+    private basePath: string = "/questions";
 
-    this.answers = db.list('answers').snapshotChanges().map((changes: any[])  => {
-              return changes.map((answerUnit: any) => ({ key: answerUnit.payload.key, ...answerUnit.payload.val() }));
+    question: AngularFireList<Question> = null;
```
Comment (FrontendMentor) [inline src/app/services/question.service.ts]: I'm not sure that you need to use `null` here and below.

File: src/app/services/question.service.ts [hunk 10]
```diff
@@ -13,35 +14,72 @@ export class QuestionService {
...
+    addQuestion(title: string, question: string, uid:string, date: Date): void {
+        this.db.list('questions').push({
+            title: title,
+            date: '' + date,
+            questionText: question,
+            uid: uid
+        })
+    }
+
+    addAnswer(answer: string, uid: string, questionKey: string, date: Date) {
```
Comment (FrontendMentor) [inline src/app/services/question.service.ts]: add return type for this method

File: src/app/services/question.service.ts [hunk 11]
```diff
@@ -13,35 +14,72 @@ export class QuestionService {
...
+            answer: answer,
+            date: '' + date,
+            qustionId: questionKey,
+            ansPersId: uid
+        })
+    }
+
+    removeItem(path: string, key: string): void {
+        
+        if (path == "questions") {
```
Comment (FrontendMentor) [inline src/app/services/question.service.ts]: Here should be `if (path === 'questions') {`

File: src/app/components/question/question.component.ts [hunk 12]
```diff
@@ -1,13 +1,37 @@
...
+        if (answer == '') {
```
Comment (FrontendMentor) [inline src/app/components/question/question.component.ts]: Let's use here `===`

File: src/app/components/questions/questions.component.ts [hunk 13]
```diff
@@ -10,17 +11,41 @@ import { QuestionComponent } from '../question/question.component';
...
+            if (this.users[user].key == id) name = this.users[user].username;
```
Comment (FrontendMentor) [inline src/app/components/questions/questions.component.ts]: Let's use here `===`

File: src/app/components/answer/answer.component.ts (hunk 4 above)
Comment (VeiveHata) [reply src/app/components/answer/answer.component.ts]: will do it

**Pull request merged**